video/image_writer.c

```diff
@@ -706,6 +707,9 @@ bool write_image(struct mp_image *image, const struct image_writer_opts *opts,
         int alpha = image->fmt.flags & MP_IMGFLAG_ALPHA;
         destfmt = alpha ? pixfmt2imgfmt(AV_PIX_FMT_YUVA420P) : IMGFMT_420P;
     }
+    if (opts->format == AV_CODEC_ID_RAWVIDEO) {
+        destfmt = IMGFMT_BGRA;
+    }
```

Comment:
I think you're better off writing a new function that dumps the mp_image contents than trying to reuse `write_lavc`